File: leetcode/20_Valid_Parentheses.java
class Solution {
    public boolean isValid(String s) {
        if(s.length()%2==1){
            return false;
        }
        int check = 0;
        while(s.length()>0){
            check = s.length();
            s=s.replace("()", "");
            s=s.replace("[]", "");
            s=s.replace("{}", "");
            if(check == s.length()){
                return false;
            }
        }
        return true;
            
    }
}

File: leetcode/20_Valid_Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        map = {")": "(", "]": "[", "}": "{"}

        for i in s:
            if i == '(' or i == '[' or i == '{':
                stack.append(i)
            elif len(stack) > 0 and map[i] == stack[-1]:
                stack.pop()
            else:
                return False

        if len(stack) == 0:
            return True

        return False
